### PR TITLE
Remove predicated load/store intrinsic

### PR DESCRIPTION
**Pull request by psuriana**
This PR deleted predicated load/store intrinsic. Store/Load IR nodes now have predicate field instead.

### CODE REVIEW

File: src/AlignLoads.cpp [hunk 1]
```diff
@@ -39,20 +39,18 @@ class AlignLoads : public IRMutator {
...
     // Rewrite a load to have a new index, updating the type if necessary.
     Expr make_load(const Load *load, Expr index) {
+        internal_assert(is_one(load->predicate)) << "Load should not be predicated.\n";
         return mutate(Load::make(load->type.with_lanes(index.type().lanes()), load->name,
-                                 index, load->image, load->param));
+                                 index, load->image, load->param, const_true(index.type().lanes())));
     }
 
-    void visit(const Call *op) {
-        // We shouldn't mess with load inside an address_of.
```
Comment (dsharletg) [inline src/AlignLoads.cpp]: I think we should still not mess with loads inside an address_of.

File: test/correctness/predicated_store_load.cpp [hunk 2]
```diff
@@ -123,8 +128,7 @@ int vectorized_dense_load_with_stride_minus_one_test() {
 
     Target target = get_jit_target_from_environment();
     if (target.features_any_of({Target::HVX_64, Target::HVX_128})) {
-        //TODO(psuriana): the hexagon test for this one is broken
-        //f.hexagon().vectorize(x, 16);
+        f.hexagon().vectorize(x, 16);
```
Comment (dsharletg) [inline test/correctness/predicated_store_load.cpp]: This should be 32 (for HVX_128).

File: src/IR.h [hunk 3]
```diff
@@ -205,7 +205,8 @@ struct Load : public ExprNode<Load> {
     // If it's a load from an image parameter, this points to that
     Parameter param;
 
-    EXPORT static Expr make(Type type, std::string name, Expr index, Buffer<> image, Parameter param);
+    EXPORT static Expr make(Type type, std::string name, Expr index, Buffer<> image,
+                            Parameter param, Expr predicate);
```
Comment (dsharletg) [inline src/IR.h]: If there were an overload/default parameter for the predicate, tons of trivial changes in this PR could be reverted...

Comment (abadams) [inline src/IR.h]: That's very dangerous - it would make it easy to silently drop a predicate in a mutator.

**Comment by dsharletg** [comment]
Thanks for the fixes!

**Pull request merged**